frame/pablo/src/lib.rs

```diff
@@ -322,6 +322,43 @@ pub mod pallet {
 		type TWAPInterval: Get<MomentOf<Self>>;
 
 		type WeightInfo: WeightInfo;
+
+		fn redeemable_assets_for_given_lp_tokens(
+			pool: PoolConfigurationOf<Self>,
+			pool_account: Self::AccountId,
+			lp_amount: Self::Balance,
+		) -> Result<(Self::Balance, Self::Balance), DispatchError> {
+			match pool {
+				PoolConfiguration::StableSwap(StableSwapPoolInfo { pair, lp_token, .. }) |
+				PoolConfiguration::ConstantProduct(ConstantProductPoolInfo {
+					pair,
+					lp_token,
+					..
+				}) => {
+					let pool_base_aum =
+						Self::Convert::convert(Self::Assets::balance(pair.base, &pool_account));
+					let pool_quote_aum =
+						Self::Convert::convert(Self::Assets::balance(pair.quote, &pool_account));
+					let lp_issued = Self::Assets::total_issuance(lp_token);
+
+					let base_amount = Self::Convert::convert(safe_multiply_by_rational(
+						Self::Convert::convert(lp_amount),
+						pool_base_aum,
+						Self::Convert::convert(lp_issued),
+					)?);
+					let quote_amount = Self::Convert::convert(safe_multiply_by_rational(
+						Self::Convert::convert(lp_amount),
+						pool_quote_aum,
+						Self::Convert::convert(lp_issued),
+					)?);
+					Ok((base_amount, quote_amount))
+				},
+
+				// for LBP just return (0,0) as it does not have LP Token
+				PoolConfiguration::LiquidityBootstrapping(_) =>
+					Ok((Self::Balance::zero(), Self::Balance::zero())),
+			}
```

Comment:
in other places when lbp is different, error returned afaik

Reply:
Yes that's true, but for LBP's remove_liquidity() lp_amount, min_base_amount, max_base_amount does not matter. So returning Error prevents call to LBP's remove_liquidity() function hence I just returned (0,0). Although I realized that I need to move check for CannotRespectMinimumRequested into individual match arms for StableSwap and ConstantProduct.

Reply:
it is not (0,0). people will burn PICA for doing nothin. may be error them for doing stupid work?

Reply:
burn PICA in RPC, or in remove_liquidity() ? I don't think it does any unnecessary work in LBP as it just return (0,0).